specifyweb/frontend/js_src/lib/components/Molecules/Dialog.tsx

```diff
@@ -47,7 +47,7 @@ const commonContainer = 'rounded resize max-w-[90%] shadow-lg shadow-gray-500';
 export const dialogClassNames = {
   fullScreen: '!transform-none !w-full !h-full',
   freeContainer: `${commonContainer} max-h-[90%]`,
-  narrowContainer: `${commonContainer} max-h-[50%] min-w-[min(20rem,90%)]
+  narrowContainer: `${commonContainer} max-h-[90%] sm:max-h-[50%] min-w-[min(20rem,90%)]
```

Comment:
(optional) use `max-h-screen` pr `max-h-[unset]` (not sure which one would work better) on small devices - max height 50% is useful on desktops so that you can see the content behind the dialog, but you can't see that much behind the dialog anyway on mobile, plus space is at a premium, so having dialog be full height would be nice.